### PR TITLE
Add initial changes to cep-47 introduction and prepare sections

### PR DESCRIPTION
**Pull request by Buddhika-Pathirathna**
### Related links
-

### Changes

Added Introduction and Preparation content to CEP-47 contract with sidebar.config file changes

### Notes
Will be completing CEP-47 tutorial deploy, implementation, and test sections with two more separate cards.

### CODE REVIEW

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+## CEP-47 Standard
+CEP47 takes full advantage of the URef feature to optimize smart contracts interactions. It also uses the Groups feature to optimize gas costs. Since this is a different type of token there are special endpoint implementations to support that unique features. You can find the [main functions](./index#cep-47-function-types) when going through the [contract file](https://github.com/casper-ecosystem/casper-nft-cep47/blob/master/cep47/bin/cep47_token.rs) inside the [CEP-47 repository](https://github.com/casper-ecosystem/casper-nft-cep47).
+
+### CEP-47 Function Types
+CEP-47 standard contains endpoints to enable the NFT functionalities. The basic attributes of tokens are addressed by *name*, *symbol*, *total_supply*, and *balance_of* endpoints.The token ownership is managed by *owner_of*, *get_token_by_index* endpoints. Token metadata is handled  through *meta*, *token_meta*, *update_token_meta* endpoints. Main behaviors of NFTs are handled through *mint*, *mint_copies*, *burn*, *transfer*, *transfer_from*, *approve*, *get_approved* endpoints.
```
Comment (deuszex) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:15]: `name`, `symbol`, `meta`, `total_supply` are details of the whole contract.
`balance_of`, `get_token_by_index` get details of tokens related to a specific account.
`owner_of`, `token_meta` are for details of a specific token.
All of the above can only be called from inside contracts, as they return data.

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+### ERC-721 Standard
+ERC-721 is the standard defined to implement the [non-fungible tokens](/docs/glossary/N#non-fungible-token) inside Ethereum blockchain. This standard defines a minimum interface a smart contract should implement to manage, own and trade the unique tokens.  
+
+## ERC-721 Implementation within Casper Network
+While [ERC-721](https://eips.ethereum.org/EIPS/eip-721) is the original name for the standard given in Ethereum blockchain, CEP-47 is the customized standard for the Casper platform. CEP-47 standard supports the NFT behaviors inside the Casper network. Casper ecosystem needs an NFT standard to enable further development of the NFT-based products. Due to the unique features within the Casper VM, the standards originate from Ethereum ecosystem are not the best choice for the Casper platform. Hence, CEP-47 is built to support these deviations and it is mainly focused on a gas-efficient design to support those unique traits.
```
[the remark this answers is not in the document]
Comment (Buddhika-Pathirathna) [reply source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:9]: > `the standards originate from Ethereum ecosystem are not the best choice for the Casper platform` is horrible phrasing, suggests that we cannot do what eth can

Thanks, @deuszex - I grabbed that from PR 47 (file - [./0047-casper-nft-protocol.md](https://github.com/casper-network/ceps/pull/47/files#) -> line 12-14 ). Could you please suggest a good alternative way to describe it or do you think it's better to remove that part?

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+## CEP-47 Standard
+CEP47 takes full advantage of the URef feature to optimize smart contracts interactions. It also uses the Groups feature to optimize gas costs. Since this is a different type of token there are special endpoint implementations to support that unique features. You can find the [main functions](./index#cep-47-function-types) when going through the [contract file](https://github.com/casper-ecosystem/casper-nft-cep47/blob/master/cep47/bin/cep47_token.rs) inside the [CEP-47 repository](https://github.com/casper-ecosystem/casper-nft-cep47).
```
[the remark this answers is not in the document]
Comment (Buddhika-Pathirathna) [reply source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:12]: This part is also taken from [Cep-47 - Casper NFT Protocol](https://github.com/casper-network/ceps/pull/47/files#) file , line numbers 16 17. Appreciate if you can re-phrase it in a better way, suggest any new idea or verify the deletion of that part. Thanks

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Introduction
+
+This tutorial takes you through the NFT implementation of the Casper Network. Before diving deep into the Casper NFT implementation, let's cover some knowledge on overall NFT design. The following sections illustrate the origin and the development of NFTs in the blockchain domain.
+
+### ERC-721 Standard
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:5]:
```suggestion
### The NFT Standard on Ethereum
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+ERC-721 is the standard defined to implement the [non-fungible tokens](/docs/glossary/N#non-fungible-token) inside Ethereum blockchain. This standard defines a minimum interface a smart contract should implement to manage, own and trade the unique tokens.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:6]:
```suggestion
[ERC-721](https://eips.ethereum.org/EIPS/eip-721) is the standard defined to implement NFTs on the Ethereum blockchain. This standard defines a minimum interface a smart contract should implement to manage, own, and trade unique tokens. 
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# Introduction
+
+This tutorial takes you through the NFT implementation of the Casper Network. Before diving deep into the Casper NFT implementation, let's cover some knowledge on overall NFT design. The following sections illustrate the origin and the development of NFTs in the blockchain domain.
+
+### ERC-721 Standard
+ERC-721 is the standard defined to implement the [non-fungible tokens](/docs/glossary/N#non-fungible-token) inside Ethereum blockchain. This standard defines a minimum interface a smart contract should implement to manage, own and trade the unique tokens.  
+
+## ERC-721 Implementation within Casper Network
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:8]:
```suggestion
## The NFT Standard on Casper
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
+# Introduction
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:1]:
```suggestion
# The NFT Standard on Casper (CEP-47)
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+This tutorial takes you through the NFT implementation of the Casper Network. Before diving deep into the Casper NFT implementation, let's cover some knowledge on overall NFT design. The following sections illustrate the origin and the development of NFTs in the blockchain domain.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:3]:
```suggestion
This tutorial takes you through the standard of implementing [non-fungible tokens](/docs/glossary/N#non-fungible-token) on the Casper Network. Before diving into the Casper NFT implementation, let's cover some knowledge on overall NFT design. The following sections illustrate the origin and the development of NFTs in the blockchain domain.
```

Comment (ipopescu) [reply source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:3]: I moved the NFT definition up, where it is first used.

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md (hunk 5 above)
Comment (ipopescu) [reply source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:6]: Please double check that links and definitions are mentioned when a term is first used (such as ERC-721, CEP-47).

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md (hunk 2 above)
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:9]:
```suggestion
[CEP-47](https://github.com/casper-ecosystem/casper-nft-cep47) is the NFT standard for the Casper blockchain, supporting the unique traits of the Casper Virtual Machine and its gas-efficient design. 
```

[the remark this answers is not in the document]
Comment (ipopescu) [reply source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:9]: I recommend simplifying this and keeping it short and sweet -- unless you can get more information from developers about the "unique traits" and describe those further. Proposed commit below.

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+### ERC-721 Standard
+ERC-721 is the standard defined to implement the [non-fungible tokens](/docs/glossary/N#non-fungible-token) inside Ethereum blockchain. This standard defines a minimum interface a smart contract should implement to manage, own and trade the unique tokens.  
+
+## ERC-721 Implementation within Casper Network
+While [ERC-721](https://eips.ethereum.org/EIPS/eip-721) is the original name for the standard given in Ethereum blockchain, CEP-47 is the customized standard for the Casper platform. CEP-47 standard supports the NFT behaviors inside the Casper network. Casper ecosystem needs an NFT standard to enable further development of the NFT-based products. Due to the unique features within the Casper VM, the standards originate from Ethereum ecosystem are not the best choice for the Casper platform. Hence, CEP-47 is built to support these deviations and it is mainly focused on a gas-efficient design to support those unique traits.
+
+## CEP-47 Standard
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:11]:
```suggestion
```

Comment (ipopescu) [reply source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:11]: Suggesting to remove the title and merge the content with the previous paragraph.

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md (hunk 3 above)
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:12]:
```suggestion
CEP-47 takes full advantage of [unforgeable references](https://casper.network/docs/design/uref) to store values and manage permissions to them. The standard also takes advantage of other access control features (such as groups). We recommend exploring the [main functions](./index#cep-47-function-types) of the [contract](https://github.com/casper-ecosystem/casper-nft-cep47/blob/master/cep47/bin/cep47_token.rs) to understand the standard further.
```

[the remark this answers is not in the document]
Comment (ipopescu) [reply source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:12]: 1. I suggest simplifying (see commit suggestion), unless you can find out more about how those optimizations can be made. However, since this is an introduction, you can talk more about the optimizations in another section.
2. Please add the Groups explanation to the glossary, or somewhere in the design section: 
"The Groups feature provides access control to a contract's runtime context. It is used only once to restrict the use of the constructor entry_point, which sets up necessary data storage in the contract context."
3. Once you have a link, please add it as a reference where you first mention "Groups".

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md (hunk 1 above)
Comment (ipopescu) [reply source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:15]: @Buddhika-Pathirathna I recommend changing this paragraph by adding a list and a short sentence about each function. Then add what @deuszex mentioned as a note.

```
The CEP-47 standard contains the following endpoints:
*name* - ...
*symbol* - ...
*total_supply* - ...
*balance_of* - ...
*owner_of* - ...
*get_token_by_index* - ...
*meta* - ...
*token_meta* - ...
*update_token_meta*  - ...
*mint* - ...
*mint_copies* - ...
*burn* - ...
*transfer* - ...
*transfer_from* - ...
*approve* - ...
*get_approved*  - ...

:::note
These functions can only be called from inside contracts, as they return data:
*name*, *symbol*, *meta*, and *total_supply* return details regarding the whole contract
*balance_of* and *get_token_by_index* retrieve details of tokens related to a specific account
*owner_of* and *token_meta* retrieve the details of a specific token
:::
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+### CEP-47 Function Types
+CEP-47 standard contains endpoints to enable the NFT functionalities. The basic attributes of tokens are addressed by *name*, *symbol*, *total_supply*, and *balance_of* endpoints.The token ownership is managed by *owner_of*, *get_token_by_index* endpoints. Token metadata is handled  through *meta*, *token_meta*, *update_token_meta* endpoints. Main behaviors of NFTs are handled through *mint*, *mint_copies*, *burn*, *transfer*, *transfer_from*, *approve*, *get_approved* endpoints.
+
+
+If you haven't read [Writing Rust Contracts on Casper](../../writing-contracts/rust.md), we recommend you start there to learn more about smart contracts.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:18]:
```suggestion
```

Comment (ipopescu) [reply source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:18]: I think it is best to remove this line.

**Review by ipopescu — CHANGES_REQUESTED** [review]
@Buddhika-Pathirathna thank you for this section! I have reviewed it and left suggested changes. Please let me know if you have questions.

File: source/docs/casper/glossary/N.md [hunk 11]
```diff
@@ -18,3 +18,6 @@ You will encounter different types of nodes on the network:
 ## Node operator {#node-operator}
 
 See [operator](O.md#operator).
+
+## Non-Fungible Token (NFT) {#non-fungible-token}
```
[the remark this answers is not in the document]
Comment (Buddhika-Pathirathna) [reply source/docs/casper/glossary/N.md:22]: Thanks a lot @Iulia for taking time to review, I will add the changes soon

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+### The NFT Standard on Ethereum
+[ERC-721](https://eips.ethereum.org/EIPS/eip-721) is the standard defined to implement NFTs on the Ethereum blockchain. This standard defines a minimum interface a smart contract should implement to manage, own, and trade unique tokens. 
+
+## The NFT Standard on Casper
+[CEP-47](https://github.com/casper-ecosystem/casper-nft-cep47) is the NFT standard for the Casper blockchain, supporting the unique traits of the Casper Virtual Machine and its gas-efficient design. 
+
+CEP-47 takes full advantage of [unforgeable references](/docs/design/uref) to store values and manage permissions to them. The standard also takes advantage of other access control features (such as [groups](/docs/glossary/G#groups)). We recommend exploring the [main functions](../cep47/#cep-47-function-types) of the [contract](https://github.com/casper-ecosystem/casper-nft-cep47/blob/master/cep47/bin/cep47_token.rs) to understand the standard further.
+
+### CEP-47 Function Types
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:13]:
```suggestion
### CEP-47 Functions
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+CEP-47 standard contains following endpoints to enable the NFT functionalities.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:14]:
```suggestion
The CEP-47 standard contains the following functions to enable NFTs. 
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+[ERC-721](https://eips.ethereum.org/EIPS/eip-721) is the standard defined to implement NFTs on the Ethereum blockchain. This standard defines a minimum interface a smart contract should implement to manage, own, and trade unique tokens. 
+
+## The NFT Standard on Casper
+[CEP-47](https://github.com/casper-ecosystem/casper-nft-cep47) is the NFT standard for the Casper blockchain, supporting the unique traits of the Casper Virtual Machine and its gas-efficient design. 
+
+CEP-47 takes full advantage of [unforgeable references](/docs/design/uref) to store values and manage permissions to them. The standard also takes advantage of other access control features (such as [groups](/docs/glossary/G#groups)). We recommend exploring the [main functions](../cep47/#cep-47-function-types) of the [contract](https://github.com/casper-ecosystem/casper-nft-cep47/blob/master/cep47/bin/cep47_token.rs) to understand the standard further.
+
+### CEP-47 Function Types
+CEP-47 standard contains following endpoints to enable the NFT functionalities. 
+The CEP-47 standard contains the following endpoints:
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:15]: Delete this line, as it seems to be duplicated.
```suggestion
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+Metadata endpoints will be used by other smart contracts
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:16]: I think this line should also be deleted.
```suggestion
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+
+### The NFT Standard on Ethereum
+[ERC-721](https://eips.ethereum.org/EIPS/eip-721) is the standard defined to implement NFTs on the Ethereum blockchain. This standard defines a minimum interface a smart contract should implement to manage, own, and trade unique tokens. 
+
+## The NFT Standard on Casper
+[CEP-47](https://github.com/casper-ecosystem/casper-nft-cep47) is the NFT standard for the Casper blockchain, supporting the unique traits of the Casper Virtual Machine and its gas-efficient design. 
+
+CEP-47 takes full advantage of [unforgeable references](/docs/design/uref) to store values and manage permissions to them. The standard also takes advantage of other access control features (such as [groups](/docs/glossary/G#groups)). We recommend exploring the [main functions](../cep47/#cep-47-function-types) of the [contract](https://github.com/casper-ecosystem/casper-nft-cep47/blob/master/cep47/bin/cep47_token.rs) to understand the standard further.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:11]:
```suggestion
The Casper NFT standard takes full advantage of [unforgeable references](/docs/design/uref) to store values and manage permissions to them. It also takes advantage of other access control features (such as [groups](/docs/glossary/G#groups)). We recommend exploring the [main functions](../cep47/#cep-47-function-types) of the [contract](https://github.com/casper-ecosystem/casper-nft-cep47/blob/master/cep47/bin/cep47_token.rs) to understand the standard further.
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+
+### CEP-47 Function Types
+CEP-47 standard contains following endpoints to enable the NFT functionalities. 
+The CEP-47 standard contains the following endpoints:
+Metadata endpoints will be used by other smart contracts
+- [*name*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L44-L47) - Returns the name of the NFT contract
+- [*symbol*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L50-L53) - Returns the symbol of the NFT contract
+- [*meta*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L56-L59) - Returns the metadata of NFT
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:19]:
```suggestion
- [*meta*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L56-L59) - Returns the metadata of the NFT contract
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+- [*total_supply*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L62-L65)- Returns the amount of issued NFTs
+- [*balance_of*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L68-L72) - Returns an amount of NFT instances the `owner` owns
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:21]:
```suggestion
- [*balance_of*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L68-L72) - Returns the amount of NFT tokens the `owner` holds
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+- [*get_token_by_index*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L75-L80) - Getter for token index. Tokens are stored by tokenId and by an index
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:22]: I don't understand this explanation. Do you mean this:

```suggestion
- [*get_token_by_index*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L75-L80) - Retrieves the NFT token at a specific index
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+- [*owner_of*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L83-L87) - Getter for the owner of a given token
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:23]:
```suggestion
- [*owner_of*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L83-L87) - Retrieves the owner of a given token
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+- [*token_meta*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L90-L94) - Getter for the metadata for a given token
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:24]:
```suggestion
- [*token_meta*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L90-L94) - Retrieves the metadata for a given token
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+- [*update_token_meta*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L97-L103) -  A function to update the metadata of a token
+- [*mint*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L106-L113) - Creates a list of tokens with the given ids and the given metas for one given recipient. tokenIds and metas are paired in order
+- [*mint_copies*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L116-L124) - Creates for an address, several tokens with the given tokenIds but with the same meta
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:27]:
```suggestion
- [*mint_copies*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L116-L124) - For a given address, this function creates several tokens with specific IDs but with the same metadata
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 23]
```diff
@@ -0,0 +1,40 @@
...
+- [*burn*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L127-L133) - Destroys the given tokens for a given account
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:28]:
```suggestion
- [*burn*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L127-L133) - Destroys the tokens in the account given
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+- [*transfer*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L136-L142) - Transfers tokens to another account
+- [*transfer_from*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L145-L152) - Transfer tokens from a given account to another account
+- [*approve*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L155-L161) - Give an account 'approved rights' to transfer tokens that you hold
+- [*get_approved*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L164-L169) - Getter function to see if the owner has given you approved rights to the required tokens
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:32]: Suggestion to write this in plain English:
```suggestion
- [*get_approved*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L164-L169) - Retrieves information about the rights to transfer tokens from another account
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 25]
```diff
@@ -0,0 +1,40 @@
...
+- [*mint_copies*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L116-L124) - Creates for an address, several tokens with the given tokenIds but with the same meta
+- [*burn*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L127-L133) - Destroys the given tokens for a given account
+- [*transfer*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L136-L142) - Transfers tokens to another account
+- [*transfer_from*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L145-L152) - Transfer tokens from a given account to another account
+- [*approve*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L155-L161) - Give an account 'approved rights' to transfer tokens that you hold
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:31]: Suggestion to write this in plain English:
```suggestion
- [*approve*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L155-L161) - Gives another account the right to transfer tokens from this account
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md [hunk 26]
```diff
@@ -0,0 +1,40 @@
...
+- [*get_token_by_index*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L75-L80) - Getter for token index. Tokens are stored by tokenId and by an index
+- [*owner_of*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L83-L87) - Getter for the owner of a given token
+- [*token_meta*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L90-L94) - Getter for the metadata for a given token
+- [*update_token_meta*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L97-L103) -  A function to update the metadata of a token
+- [*mint*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L106-L113) - Creates a list of tokens with the given ids and the given metas for one given recipient. tokenIds and metas are paired in order
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/index.md:26]:
```suggestion
- [*mint*](https://github.com/casper-ecosystem/casper-nft-cep47/blob/09b40b0caf4cfc6f73d1e5f7d5b9c868228f7621/cep47/bin/cep47_token.rs#L106-L113) - Creates a list of tokens for a specific recipient, given the token IDs and their metadata, paired in order
```

**Review by ipopescu — COMMENTED** [review]
@Buddhika-Pathirathna thank you for the updates! Almost there!
I have some final suggestions and if you agree with them, please push the new commits and I will approve the PR. Thanks so much!

File: source/docs/casper/glossary/G.md [hunk 27]
```diff
@@ -14,6 +14,9 @@ Gas is the virtual currency for calculating the cost of transaction execution. T
 
 The state of the virtual machine at the beginning of the blockchain.
 
+## Groups {#groups}
+The Groups feature provides access control to a contract's runtime context. It is used only once to restrict the use of the constructor entry_point, which sets up necessary data storage in the contract context.
```
Comment (deuszex) [inline source/docs/casper/glossary/G.md:18]: contracts*

Comment (Buddhika-Pathirathna) [reply source/docs/casper/glossary/G.md:18]: I would like to get @ipopescu 's input for this.

Comment (ipopescu) [reply source/docs/casper/glossary/G.md:18]: Based on our group discussion, providing a wider definition and clarifying the possessive form for the contract.
```suggestion
The *user groups* feature provides access control to the entry points of a contract by creating a new user group for that contract (versioned or not). This feature restricts the use of the constructor entry_point, which sets up the necessary data storage in the runtime context that belongs to the contract. Here is how it works:
- User groups associate a set of URefs with a label.
- The entry points on a contract can accept a list of labels
- The runtime checks that a URef from at least one of the allowed groups is present in the caller's context before execution. 
```

**Review by ipopescu — APPROVED** [review]
Good to go! Thanks, @Buddhika-Pathirathna!

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/implementation.md [hunk 28]
```diff
@@ -0,0 +1 @@
+# Contract Implementation
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/implementation.md:1]:
```suggestion
# Contract Implementation

Coming soon!
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/deploy.md [hunk 29]
```diff
@@ -0,0 +1 @@
+# Contract Deployment
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/deploy.md:1]:
```suggestion
# Contract Deployment

Coming soon!
```

File: source/docs/casper/dapp-dev-guide/tutorials/cep47/test.md [hunk 30]
```diff
@@ -0,0 +1 @@
+# Testing Framework
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/cep47/test.md:1]:
```suggestion
# Testing Framework

Coming soon!
```

**Review by ipopescu — COMMENTED** [review]
Adding a note on the blank pages as this may go out live if we publish it before pr 206 (this depends on how 206 goes).

**Comment by ipopescu** [comment]
Great work, @Buddhika-Pathirathna! This has been merged.
